api/common_types.go

```diff
@@ -8,14 +8,14 @@ import (
 // KubernetesConfig will be the JSON struct for Basic Redis Config
 // +k8s:deepcopy-gen=true
 type KubernetesConfig struct {
-	Image                  string                           `json:"image"`
-	ImagePullPolicy        corev1.PullPolicy                `json:"imagePullPolicy,omitempty"`
-	Resources              *corev1.ResourceRequirements     `json:"resources,omitempty"`
-	ExistingPasswordSecret *ExistingPasswordSecret          `json:"redisSecret,omitempty"`
-	ImagePullSecrets       *[]corev1.LocalObjectReference   `json:"imagePullSecrets,omitempty"`
-	UpdateStrategy         appsv1.StatefulSetUpdateStrategy `json:"updateStrategy,omitempty"`
-	Service                *ServiceConfig                   `json:"service,omitempty"`
-	IgnoreAnnotations      []string                         `json:"ignoreAnnotations,omitempty"`
+	Image              string                           `json:"image"`
+	ImagePullPolicy    corev1.PullPolicy                `json:"imagePullPolicy,omitempty"`
+	Resources          *corev1.ResourceRequirements     `json:"resources,omitempty"`
+	ExistingAuthSecret *ExistingAuthSecret              `json:"redisSecret,omitempty"`
```

Comment:
Changing the field name would produce a breaking change here. 
we are good with adding field but don't rename any field